From 87eaa589e97d62fe7609fc42594aeca5b9ff9365 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Mon, 6 Dec 2010 14:36:27 +0000
Patch-mainline: 2.6.38
References: fate#310916
Git-commit: 0be732841fb925b6f1242211ea211c022b6ac26c
Subject: [PATCH 1928/2588] drm/i915: Wait for the bo if a display flip is
 pipelined on the other ring

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit 0be732841fb925b6f1242211ea211c022b6ac26c)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_gem.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 24912d5..ff0fc62 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -2958,7 +2958,7 @@ i915_gem_object_set_to_display_plane(struct drm_i915_gem_object *obj,
 		return ret;
 
 	/* Currently, we are always called from an non-interruptible context. */
-	if (!pipelined) {
+	if (pipelined != obj->ring) {
 		ret = i915_gem_object_wait_rendering(obj, false);
 		if (ret)
 			return ret;
-- 
1.7.6

